### ISSUE TITLE
Not all JSHint warnings showing when indentation is set to tabs

### ISSUE THREAD

**Issue by raduserbanescu** [AtomLinter/linter-jshint#416]
When a file is using **tabs** for indentation, not all JSHint warnings are shown by the `linter-jshint` package. If the same code is written in a file which uses **spaces** for indentation everything works as normal.

There are no errors in Atom's console.

Directly linting the code using the `jshint` npm package works just fine in **both** cases.

I made a simple example of code that usually fails to lint correctly. (see below)  
Screenshots: <http://imgur.com/a/SPMqL>

Maybe related to Issue #403

### Atom

```
Atom    : 1.17.2 (tested on a colleague's 1.18 version an it's still the same)
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0
```

### Packages

```
autocomplete-paths@1.0.5
busy-signal@1.4.3
clipboard-plus@0.5.1
color-picker@2.2.5
docblockr@0.10.5
editorconfig@2.2.2
emmet@2.4.3
file-icons@2.1.7
highlight-selected@0.13.1
intentions@1.1.2
less-autocompile@1.0.0
linter@2.2.0
linter-eslint@8.2.1
linter-jshint@3.1.2
linter-jsonlint@1.3.0
linter-stylelint@3.10.0
linter-ui-default@1.6.2
markdown-scroll-sync@2.1.2
minimap@4.28.2
minimap-git-diff@4.3.1
minimap-highlight-selected@4.6.1
project-manager@3.3.5
sublime-block-comment@0.5.1
Sublime-Style-Column-Selection@1.7.4
todo-show@1.11.0
```

### .jshintrc

```json
{
  "bitwise": true,
  "curly": true,
  "eqeqeq": true,
  "esversion": 6,
  "forin": true,
  "freeze": true,
  "futurehostile": true,
  "latedef": true,
  "noarg": true,
  "nocomma": true,
  "nonbsp": true,
  "nonew": true,
  "singleGroups": false,
  "strict": "global",
  "undef": true,
  "unused": true,
  "varstmt": false,

  "immed": true,
  "newcap": true,
  "noempty": true,

  "laxbreak": true,

  "browser": true,
  "jasmine": true,

  "globals": {
    "angular": false
  }
}
```

### Atom config.js

```cson
linter:
  lintOnChange: false
  lintOnOpen: false
  lintPreviewTabs: false
"linter-jshint":
  disableWhenNoJshintrcFileInPath: true
"linter-ui-default":
  showPanel: true
```

### Tabs indentation

```js
'use strict';

angular.module('app').controller('randomController', [
	'$scope',
	'$log',
	'$http',
	function randomController(
		$scope,
		$log,
		$http
	) {
		$scope.language = 'en';
		$scope.var1 = 1;
		$scope.var2 = 2;

		function fun1() {
			// Do something here
		}

		function fun2() {
			// Do something here
		};

		function fun3(param) {
			// Do something here
		};

		function main() {
			if ($scope.var1 === 1) {
					if ($scope.var2 === 2) {
							if ($scope.language === language.current) {
								fun1();
							}
							else {
								fun3();
							}
					}
					else {
						fun2();
					}
			}
		}

		main();
	}
]);
```

### Spaces indentation

```js
'use strict';

angular.module('app').controller('randomController', [
    '$scope',
    '$log',
    '$http',
    function randomController(
        $scope,
        $log,
        $http
    ) {
        $scope.language = 'en';
        $scope.var1 = 1;
        $scope.var2 = 2;

        function fun1() {
            // Do something here
        }

        function fun2() {
            // Do something here
        };

        function fun3(param) {
            // Do something here
        };

        function main() {
            if ($scope.var1 === 1) {
                    if ($scope.var2 === 2) {
                            if ($scope.language === language.current) {
                                fun1();
                            }
                            else {
                                fun3();
                            }
                    }
                    else {
                        fun2();
                    }
            }
        }

        main();
    }
]);
```

**Comment by Arcanemagus** [comment]
It looks like virtually every line it is returning has an invalid point, and unfortunately the code that was supposed to report this isn't working correctly.

**Comment by Arcanemagus** [comment]
Fix published in v3.1.3, however those messages will now simply show up as invalid points from JSHint asking you to report them till https://github.com/jshint/jshint/issues/3151 gets fixed.

**Comment by raduserbanescu** [comment]
Thank you for looking into this and solving the problem at your end. I will keep an eye on the JSHint issue.

Now that you have mentioned it, I see that in the command line the reported line and column are erroneous for the file with tab indentation.
